spec/02-integration/05-proxy/01-router_spec.lua

```diff
@@ -134,6 +134,26 @@ describe("Router", function()
       assert.equal("api-1", res.headers["kong-api-name"])
     end)
 
+    it("routes by method-only if no Host header present", function()
+      -- a very limited HTTP client for sending requests without Host
+      -- based on the source of https://github.com/pintsized/lua-resty-http
+      local sock = ngx.socket.tcp()
+      finally(function() sock:close() end)
+      local ok, err = sock:connect(helpers.test_conf.proxy_ip,
+        helpers.test_conf.proxy_port)
+      assert(ok, err)
+      local req = "GET /get HTTP/1.0\r\nKong-Debug: 1\r\n\r\n"
+      sock:send(req)
+      local line, err = sock:receive("*l")
+      assert(line, err)
+      local status = tonumber(string.sub(line, 10, 12))
+      assert.equal(status, 200)
```

Comment:
The arguments need to be reversed here: it will produce a confusing error message if this fails (expected values come first in luassert, which is the library providing those extended assertions)